packages/core/src/asset/AssetPromise.ts

```diff
@@ -102,12 +120,26 @@ export class AssetPromise<T> implements PromiseLike<T> {
    * @param callback
    * @returns AssetPromise
    */
-  onProgress(callback: (progress: number) => void): AssetPromise<T> {
-    if (this._progress) {
-      callback(this._progress);
+  onProgress(
+    itemsCallback?: (loaded: number, total: number) => void,
+    detailsCallback?: (url: string, loaded: number, total: number) => void
```

Comment:
`detailsCallback` should changed to `detailCallback`?

Reply:
OK，maybe `onTaskComplete`  and `onTaskDetail` is better